tests/unit/test_charm.py

```diff
@@ -51,6 +51,26 @@ def test_invalid_repository(harness: ops.testing.Harness[GithubProfilesAutomator
     )
 
 
+@pytest.mark.parametrize(
+    "principal_key",
+    [
+        "kfp-ui-principal",
+        "istio-ingressgateway-principal",
+    ],
+)
+def test_empty_kfp_principal(
```

Comment:
Since we're checking both principals, a different name might be more accurate:
```suggestion
def test_empty_principal_config(
```